base/applications/rapps/geninst.cpp

```diff
@@ -0,0 +1,813 @@
+/*
+ * PROJECT:     ReactOS Applications Manager
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Act as installer and uninstaller for applications distributed in archives
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ */
+
+#include "defines.h"
+#include <shlobj.h>
+#include <shlwapi.h>
+#include <setupapi.h>
+#include <commctrl.h>
+#include "resource.h"
+#include "appdb.h"
+#include "appinfo.h"
+#include "misc.h"
+#include "configparser.h"
+#include "unattended.h"
+
+#include "minizip/ioapi.h"
+#include "minizip/iowin32.h"
+#include "minizip/unzip.h"
+extern "C" {
+    #include "minizip/ioapi.c"
+    #include "minizip/iowin32.c"
+    #include "minizip/unzip.c"
+}
```

Comment:
What is the problem that makes the zipfldr unusable for this?

Reply:
I don't remember (I wrote this code back in November last year) but it is certainly not compatible with Windows.

Reply:
In what aspect is it not compatible with Windows?

Reply:
ROS ZipFldr does not support returning an IStream in IShellFolder::BindToObject (IIRC). Nor does it provide an IDataObject with access to the file content.